modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -231,7 +236,16 @@ def _column_widths(self):
         """
         if self._column_widths_cache is None:
             if len(self._partitions) > 0:
-                self._column_widths_cache = [obj.width() for obj in self._partitions[0]]
+                partitions = self._partitions[0]
+                if all(map(lambda t: t._width_cache is not None, partitions)):
+                    self._column_widths_cache = [obj.width() for obj in partitions]
+                else:
+                    widths = [
+                        obj.apply(lambda df: len(df.columns)) for obj in partitions
+                    ]
+                    self._column_widths_cache = (
+                        self._partition_mgr_cls.get_objects_from_partitions(widths)
+                    )
```

Comment:
shouldn't we handle this at the PartitionClass level?

Reply:
the code also makes to recompute cache for ALL of the partitions if only ONE of them has missing cache